lib/init.php

```diff
@@ -188,6 +188,30 @@ function register_site_icon_url( $response ) {
 
 add_filter( 'rest_index', 'register_site_icon_url' );
 
+/**
+ * Exposes the site logo to the Gutenberg editor through the WordPress REST
+ * API. This is used for fetching this information when user has no rights
+ * to update settings.
+ *
+ * @since 10.9
+ *
+ * @param WP_REST_Response $response Response data served by the WordPress REST index endpoint.
+ * @return WP_REST_Response
+ */
+function register_site_logo_to_rest_index( $response ) {
+	$data              = $response->data;
+	$logo_id           = get_theme_mod( 'custom_logo' );
```

Comment:
I would really appreciate a review from @aristath since he has worked on the `theme_mod` refactoring here: https://github.com/WordPress/gutenberg/pull/31994.

Is there anything missing for the syncing process?

Reply:
Looks OK to me... I don't see anything missing. I'd say adding it to `rest_index` is a rather elegant solution :+1: